app/src/main/java/org/mozilla/fenix/collections/SaveCollectionListAdapter.kt

```diff
@@ -81,31 +82,33 @@ class CollectionViewHolder(
 
     fun bind(collection: TabCollection) {
         this.collection = collection
-        view.collection_item.text = collection.title
-
-        val hostNameList = collection.tabs.map { it.url.urlToTrimmedHost().capitalize() }
-
-        var tabsDisplayed = 0
-        val tabTitlesList = hostNameList.joinToString(", ") {
-            if (it.length > maxTitleLength) {
-                it.substring(
-                    0,
-                    maxTitleLength
-                ) + "..."
-            } else {
-                tabsDisplayed += 1
-                it
+        launch(Dispatchers.Main) {
+            view.collection_item.text = collection.title
+
+            val hostNameList = collection.tabs.map { it.url.urlToTrimmed(view.context).capitalize() }
+
+            var tabsDisplayed = 0
+            val tabTitlesList = hostNameList.joinToString(", ") {
+                if (it.length > maxTitleLength) {
+                    it.substring(
+                        0,
+                        maxTitleLength
+                    ) + "..."
+                } else {
+                    tabsDisplayed += 1
+                    it
+                }
             }
+            view.collection_description.text = tabTitlesList
+
+            view.collection_icon.setColorFilter(
+                ContextCompat.getColor(
+                    view.context,
+                    getIconColor(collection.id)
+                ),
+                android.graphics.PorterDuff.Mode.SRC_IN
+            )
         }
-        view.collection_description.text = tabTitlesList
-
-        view.collection_icon.setColorFilter(
-            ContextCompat.getColor(
-                view.context,
-                getIconColor(collection.id)
-            ),
-            android.graphics.PorterDuff.Mode.SRC_IN
-        )
     }
 
     @Suppress("ComplexMethod", "MagicNumber")
```

Comment:
Hmm, ok. 

Not part of this PR, but I think an array of colors and a modulo index would've been better here, I think.

Reply:
This reminds me that we talked about providing this in the AC API, but I actually never implemented that.

Also `id % 4` should never hit the case for `4` and `else` - so I assume `yellow` and `white` are not actually used right now?

Reply:
Good catch, @pocmo--sometimes my brain forgets modulo math 🙄. I've updated it to be 5, but yes, white should never be used.